### PR TITLE
Various fixes and improvements - no new feature

### PR DESCRIPTION
**Pull request by ttngu207**
- improvement - clustering step, update duration for "median_subtraction" step
- bugfix - handles single probe recording in "Neuropix-PXI" format
- improvement - safeguard in creating/inserting probe types upon probe activation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,12 @@
 Observes [Semantic Versioning](https://semver.org/spec/v2.0.0.html) standard and
  [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) convention.
 
+## [Unreleased]
```
Comment (CBroz1) [inline CHANGELOG.md:6]:
```suggestion
## [0.2.1] - Unreleased
```

File: element_array_ephys/readers/kilosort_triggering.py [hunk 2]
```diff
@@ -477,7 +479,8 @@ def _get_raw_data_filepaths(self):
                     for line in f.readlines():
                         if (line.startswith('ecephys spike sorting: median subtraction module')
                                 and previous_line.startswith('Total processing time:')):
-                            self._median_subtraction_finished = True
+                            duration = int(re.search('\d+\.?\d+', previous_line).group())
```
Comment (CBroz1) [inline element_array_ephys/readers/kilosort_triggering.py:482]: Please add a comment explaining this regex

File: element_array_ephys/readers/kilosort_triggering.py [hunk 3]
```diff
@@ -347,7 +347,7 @@ def __init__(self, npx_input_dir: str, ks_output_dir: str,
         self._json_directory = self._ks_output_dir / 'json_configs'
         self._json_directory.mkdir(parents=True, exist_ok=True)
 
-        self._median_subtraction_finished = False
+        self._median_subtraction_status = {}
```
Comment (CBroz1) [inline element_array_ephys/readers/kilosort_triggering.py:350]: Reviewing the docstrings across readers, they appear to be in Sphinx format. They would need to be in Google format for proper [docs rendering](https://datajoint.com/docs/elements/element-array-ephys/0.2/api/element_array_ephys/readers/kilosort/) (vs. [dlc API](https://datajoint.com/docs/elements/element-deeplabcut/0.2/api/element_deeplabcut/readers/dlc_reader/))

**Review by CBroz1 — APPROVED** [review]
Adding some stylistics suggestions

File: element_array_ephys/readers/kilosort_triggering.py [hunk 4]
```diff
@@ -428,11 +428,13 @@ def run_modules(self):
             if module_status['completion_time'] is not None:
                 continue
 
-            if module == 'median_subtraction' and self._median_subtraction_finished:
-                self._update_module_status(
-                    {module: {'start_time': datetime.utcnow(),
-                              'completion_time': datetime.utcnow(),
-                              'duration': 0}})
+            if module == 'median_subtraction' and self._median_subtraction_status:
```
[the remark this answers is not in the document]
Comment (ttngu207) [reply element_array_ephys/readers/kilosort_triggering.py:431]: Agreed, let's do that in the next PR - also need to check for datajoint version before using `dj.logger` (older DataJoint version does not have it)

File: element_array_ephys/readers/kilosort_triggering.py (hunk 3 above)
Comment (ttngu207) [reply element_array_ephys/readers/kilosort_triggering.py:350]: Agreed, will rework docs into Google format in the next PR

File: element_array_ephys/readers/kilosort_triggering.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ttngu207) [reply element_array_ephys/readers/kilosort_triggering.py:431]: For these few, I think it's more readable to keep the default values in place, at the `.get()`

File: element_array_ephys/readers/openephys.py [hunk 5]
```diff
@@ -94,14 +94,14 @@ def load_probe_data(self):
...
-                        else:
-                            probe_indices = range(len(processor['EDITOR']['PROBE']))
+                        editor_probe_key = 'PROBE'
                     elif processor['@pluginName'] == 'Neuropix-PXI':
-                        probe_indices = range(len(processor['EDITOR']['NP_PROBE']))
+                        editor_probe_key = 'NP_PROBE'
                     else:
                         raise NotImplementedError
+                    
+                    probe_indices = (0,) if isinstance(processor['EDITOR'][editor_probe_key], dict) else range(len(processor['EDITOR'][editor_probe_key]))
```
[the remark this answers is not in the document]
Comment (ttngu207) [reply element_array_ephys/readers/openephys.py:103]: BLACK formatting applied across the repo

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,12 @@
 Observes [Semantic Versioning](https://semver.org/spec/v2.0.0.html) standard and
  [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) convention.
 
+## [0.2.1] - Unreleased
+
++ Update - clustering step, update duration for "median_subtraction" step
++ Bugfix - handles single probe recording in "Neuropix-PXI" format
++ Update - safeguard in creating/inserting probe types upon probe activation
+
```
[the remark this answers is not in the document]
Comment (ttngu207) [reply CHANGELOG.md:11]: Done

File: element_array_ephys/readers/kilosort_triggering.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (CBroz1) [reply element_array_ephys/readers/kilosort_triggering.py]: This is the syntax I use there
```python
import logging
logger = logging.getLogger("datajoint")
```
As I understand it, if `datajoint` doesn't exist as a logger, [getLogger](https://docs.python.org/3/library/logging.html#logging.getLogger) will return the default root logger.

File: CHANGELOG.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (CBroz1) [reply CHANGELOG.md:11]: We're missing a link to `0.1.0b4`
```
[0.1.0b4]: https://github.com/datajoint/element-array-ephys/releases/tag/0.1.0b4
```

**Review by CBroz1 — APPROVED** [review]
Thanks @ttngu207 - There are a couple unresolved comments. If we want to merge without addressing these, lmk.

File: CHANGELOG.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (ttngu207) [reply CHANGELOG.md:11]: Done

File: element_array_ephys/readers/kilosort_triggering.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ttngu207) [reply element_array_ephys/readers/kilosort_triggering.py]: Let's save docstring revamp (Google style) for the next PR(s)

**Comment by ttngu207** [comment]
> Thanks @ttngu207 - There are a couple unresolved comments. If we want to merge without addressing these, lmk.

I've responded to the comments, if all good we can merge

**Review by CBroz1 — APPROVED** [review]

**Pull request merged**